expr/func_groupbytags.go

```diff
@@ -59,7 +62,7 @@ func (s *FuncGroupByTags) Exec(cache map[Req][]models.Series) ([]models.Series,
 	if !useName {
 		// if all series have the same name, name becomes one of our tags
 		for _, serie := range series {
-			thisName := strings.Split(serie.Target, ";")[0]
+			thisName := serie.Tags["name"]
```

Comment:
right on.
it is probably not clear, but it looks like we do try to assure the name tag is always set for any series (FuncGet sets it initially and any other function that creates new series should set it as well I think)